simulators/ethereum/engine/helper.go

```diff
@@ -314,6 +315,156 @@ func (customData *CustomPayloadData) String() string {
 	return strings.Join(customFieldsList, ", ")
 }
 
+type InvalidPayloadField string
+
+const (
+	InvalidParentHash           InvalidPayloadField = "ParentHash"
+	InvalidStateRoot                                = "StateRoot"
```

Comment:
BTW, the constants here will not have the type you expect. While `InvalidParentHash` has type `InvalidPayloadField`, the other constants have type string. See https://go.dev/play/p/tyAGBY8nkuS for an example.

You can fix this by listing the type in every line. We usually do it like this:

```
const (
      InvalidParentHash = InvalidPayloadField("ParentHash")
      ...
)
```